File: Прога/Лаб.7/cabbage/SelectionSort.java
package cabbage;

public class SelectionSort implements SortChooser {
	public void sort(float[] table) {
		int n = table.length;
		for (int i = 0; i < n-1; i++) {
			int min_idx = i;
			for (int j = i+1; j < n; j++) {
				if (table[j] < table[min_idx]) {
					min_idx = j;
				}
			}
			float temp = table[min_idx]; 
			table[min_idx] = table[i]; 
			table[i] = temp; 
		}
	}
}

File: Year 1/Semester 1/Algorithms/Lab 3/SelectionSort.py
import time

def SelectionSort( itemsList ):
    start = time.time()

    n = len( itemsList )
    for i in range( n - 1 ): 
        minValueIndex = i
        for j in range( i + 1, n ):
            if itemsList[j] < itemsList[minValueIndex] :
                minValueIndex = j
        if minValueIndex != i :
            temp = itemsList[i]
            itemsList[i] = itemsList[minValueIndex]
            itemsList[minValueIndex] = temp

    end = time.time()
    finalTime = 1000 * (end - start)
    estimatedTime = "Selection sort for " + str(n) + " elements took " + str(round(finalTime, 3)) + " miliseconds"

    return estimatedTime